contracts/example/ZoraWrapper/ZoraWrapper.sol

```diff
@@ -0,0 +1,245 @@
+     *  - caller is not the contract owner
+     *
+     * @param _tokenId The token id.
+     */
+    function wrap(uint256 _tokenId) external {
+        // Transfer voucher to this contract
+        IERC721(voucherAddress).transferFrom(msg.sender, address(this), _tokenId);
+
+        // Mint to itself, so it can be used with Zora Auction House
+        _mint(address(this), _tokenId);
+
+        // Approves original token owner to operate on wrapped token
+        _approve(msg.sender, _tokenId);
+    }
+
+    /**
+     * @notice Unwraps the voucher, transfer true voucher to owner and funds to the protocol.
+     *
+     * Reverts if:
+     *  - wrapped voucher is not owned by the seller and the caller is not the protocol
+     *  - wrapped voucher is owned by the seller and the caller is not the owner of this contract or protcol
+     *
+     * @param _tokenId The token id.
+     */
+    function unwrap(uint256 _tokenId) external {
+        address wrappedVoucherOwner = ownerOf(_tokenId);
+
+        // Either contract owner or protocol can unwrap
+        // If contract owner is unwrapping, this is equivalent to canceled auction
+        require(
+            msg.sender == protocolAddress || wrappedVoucherOwner == msg.sender,
+            "ZoraWrapper: Only owner or protocol can unwrap"
+        );
+
+        // If some token price is not know yet, update it now
+        if (pendingTokenId != 0) updatePendingTokenPrice(pendingTokenId);
+
+        uint256 priceToPay = price[_tokenId];
+
+        // Delete price and pendingTokenId to prevent reentrancy
+        delete price[_tokenId];
+        delete pendingTokenId;
+
+        // transfer voucher to voucher owner
+        IERC721(voucherAddress).safeTransferFrom(address(this), wrappedVoucherOwner, _tokenId);
+
+        // Transfer token to protocol
+        if (priceToPay > 0) {
+            // No need to handle native separately, since Zora Auction House always sends WETH
+            IERC20(cachedExchangeToken[_tokenId]).safeTransfer(protocolAddress, priceToPay);
+        }
+
+        delete cachedExchangeToken[_tokenId]; // gas refund
+
+        // Burn wrapped voucher
+        _burn(_tokenId);
+    }
+
+    /**
+     * @notice Handle transfers out of Zora Auction House.
+     *
+     * @param _from The address of the sender.
+     * @param _to The address of the recipient.
+     * @param _tokenId The token id.
+     */
+    function _beforeTokenTransfer(
+        address _from,
+        address _to,
+        uint256 _tokenId
+    ) internal virtual override(ERC721) {
+        if (_from == zoraAuctionHouseAddress && _to != address(this)) {
+            // Auction is over, and wrapped voucher is being transferred to voucher owner
+            // If recipient is address(this), it means the auction was canceled and price updating can be skipped
+
+            // If some token price is not know yet, update it now
+            if (pendingTokenId != 0) updatePendingTokenPrice(pendingTokenId);
```

Comment:
there is no need to pass `pendingTokenId` as a parameter

Reply:
Removed.